src/builtins/core/date.rs

```diff
@@ -139,15 +139,9 @@ impl core::fmt::Display for PlainDate {
     }
 }
 
-impl Ord for PlainDate {
-    fn cmp(&self, other: &Self) -> core::cmp::Ordering {
-        self.iso.cmp(&other.iso)
-    }
-}
-
 impl PartialOrd for PlainDate {
     fn partial_cmp(&self, other: &Self) -> Option<core::cmp::Ordering> {
-        Some(self.cmp(other))
+        Some(self.iso.cmp(&other.iso))
```

Comment:
Shouldn't this check that both have the same calendar first?

Reply:
On `PartiaOrd`? Not if we're going based off [specification](https://tc39.es/proposal-temporal/#sec-temporal.plaindate.compare). We can check calendar and return `None`, but we'll end up failing [`intl402` tests](https://github.com/tc39/test262/blob/main/test/intl402/Temporal/PlainDate/compare/exhaustive.js).

Reply:
Uhhh that's awkward...
The thing is that we cannot break things like `a <= b <=> a < b || a == b`, or some libraries depending on that property could cause hard to debug problems...

I'd just implement the operations as methods then. That way we're not involuntarily breaking downstream code, and we can document the spec behaviour on the methods.

Reply:
I can that's probably the best approach.

I do think it's fine to implement though as is if we wanted since conditions 2-5 are upheld by PartialOrd. The question mostly seems to be around the language "a == b if and only if partial_cmp(a, b) == Some(Equal)." Depending how "only if" is applied, I think this is still technically upheld because we'd be returning a subset of values. We never return a value where partial_cmp is not Some(Equal).

But for safety, it's probably best to just use methods

Reply:
>  "a == b if and only if partial_cmp(a, b) == Some(Equal)."

This property is the same as saying `a != b iff partial_cmp(a,b) != Some(Equal)`, which with the current implementation  would be broken by two dates that are the same ISO date with different calendars, so you can definitely break that invariant trivially.

Reply:
Yeah :confused: that's true.

Methods are the best way to go.